### ISSUE TITLE
Handle Camunda Rest Failures

### ISSUE THREAD

**Issue by sreagleeye** [AMalininHere/camunda-worker-dotnet#1]
I came across an issue when trying to complete a Service Task. Ultimately, the process I was trying to complete errors out during a downstream Script Task. This causes Camunda to issue a 500 as the response to the request `external-task/{taskId}/complete`.

This gave me some confusion because the task seems to complete without issue. However, the process would still be waiting on the same Service Task. Tracking this down, it looks like `ExternalTaskCamundaClient` is not handling failed requests. 

```
private async Task<HttpResponseMessage> SendRequest(string path, object body, CancellationToken cancellationToken)
{
    var basePath = _httpClient.BaseAddress.AbsolutePath.TrimEnd('/');
    var requestPath = $"{basePath}/{path}";
    var response = await _httpClient.PostJsonAsync(requestPath, body, cancellationToken);
    return response;
}
```

I know this is still version 0, but I just wanted to bring it to your attention in case you were not aware.

I love this library! Keep up the good work!

**Comment by AMalininHere** [comment]
@sreagleeye Thank you for your issue. It will be fixed soon.

**Comment by AMalininHere** [comment]
@sreagleeye Could you provide bpmn and handler to check that case?

**Comment by sreagleeye** [comment]
[CompletionFailure.bpmn.txt](https://github.com/AMalininHere/camunda-worker-dotnet/files/3075623/CompletionFailure.bpmn.txt)

```
[HandlerTopics("completesuccessfully", LockDuration = 10_000)]
public class CompleteSuccessfullyTaskHandler : IExternalTaskHandler
{
    public async Task<IExecutionResult> Process(ExternalTask externalTask)
    {
        return await Task.FromResult(new CompleteResult(new Dictionary<string, Variable>()));
    }
}
```

**Comment by AMalininHere** [comment]
It seems to be fixed in `0.5.1`. Can you check it?

**Comment by sreagleeye** [comment]
This looks good! Once the ExternalTaskHandler completes, I am getting an incident on the service task. The stack trace for the incident is showing the error message from the failed request.
